service/storage_service.cc

```diff
@@ -938,6 +931,45 @@ class topology_coordinator {
         co_return std::pair{gen_uuid, std::move(gen_mutations)};
     }
 
+    // Broadcasts all mutations returned from prepare_new_cdc_generation_data except the last one.
+    // Each mutation is sent in separate raft command and is retried as long as group0_concurrent_modification occurs.
+    // It takes group0_guard and creates a new one for each mutation. Commands are sent sequentially.
+    // Returns the generation's UUID, last created guard and last mutation which
+    // will be sent with additional topology data by the caller.
+    future<std::tuple<utils::UUID, group0_guard, canonical_mutation>> prepare_and_broadcast_cdc_generation_data(
+            locator::token_metadata_ptr tmptr, group0_guard guard, std::optional<bootstrapping_info> binfo) {
+        auto [gen_uuid, gen_mutations] = co_await prepare_new_cdc_generation_data(tmptr, guard, binfo);
+
+        if (gen_mutations.empty()) {
+            on_internal_error(slogger, "cdc_generation_data: gen_mutations is empty");
+        }
+
+        std::vector<canonical_mutation> updates{gen_mutations.begin(), gen_mutations.end()};
+
+        for (size_t i{0}; i < updates.size() - 1; ++i) {
+            bool done{false};
+
+            auto const reason = format(
+                "insert CDC generation data (UUID: {}), part {}/{}", gen_uuid, i + 1, updates.size());
+
+            while (!done) {
+                try {
+                    slogger.trace("raft topology: do update {} reason {}", updates[i], reason);
+                    write_mutations change{{std::move(updates[i])}};
+                    group0_command g0_cmd = _group0.client().prepare_command(std::move(change), guard, reason);
+                    co_await _group0.client().add_entry(std::move(g0_cmd), std::move(guard));
+                    done = true;
+                } catch (group0_concurrent_modification&) {
```

Comment:
Please write a comment providing the reason why we catch it here instead of letting it propagate to the top-level of coordinator code (e.g. we don't want to start from scratch if there are N commands to send and there's a concurrent schema change when we're nearly done - you should calculate what N will be in large clusters).

Reply:
Done.